libbeat/common/string.go

```diff
@@ -0,0 +1,21 @@
+package common
+
+import (
+	"reflect"
+	"unsafe"
+)
+
+type NetString []byte
+
+// implement encoding.TextMarshaller interface to treat []byte as raw string
+// by other encoders/serializers (e.g. JSON)
+
+func (n NetString) MarshalText() ([]byte, error) {
+	return n, nil
+}
+
+func (n NetString) UnsafeString() string {
+	bh := (*reflect.SliceHeader)(unsafe.Pointer(&n))
+	sh := reflect.StringHeader{Data: bh.Data, Len: bh.Len}
+	return *(*string)(unsafe.Pointer(&sh))
```

Comment:
Not sure we want to get ourselves in the Unsafe territory. How much do we win in terms of performance? Perhaps we can find another way to avoid the copy by refactoring at an upper layer.